.github/workflows/node.js.yml

```diff
@@ -0,0 +1,24 @@
+name: Node CI
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.12.0, 10.x, 12.x, 14.x]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm install
```

Comment:
We could cache dependencies to have better performances in CI, something like that :
```yaml
- name: 'Cache dependencies'
  uses: 'actions/cache@v2'
  with:
    path: |
      **/node_modules
    key: ${{ runner.os }}-${{ hashFiles('**/package.json') }}
```

Reply:
The `actions/cache` project does not recommend that we use `**/node_modules`.
They recommend we do this:

```yaml
- uses: actions/cache@v2
  with:
    path: ~/.npm
    key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
    restore-keys: |
      ${{ runner.os }}-node-
```

Source: https://github.com/actions/cache/blob/main/examples.md#node---npm

Reply:
And the GitHub docs recommend this:

> This example creates a new cache when the packages in `package-lock.json` file change, or when the runner's operating system changes.

```yaml
    - name: Cache node modules
      uses: actions/cache@v2
      env:
        cache-name: cache-node-modules
      with:
        # npm cache files are stored in `~/.npm` on Linux/macOS
        path: ~/.npm
        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
        restore-keys: |
          ${{ runner.os }}-build-${{ env.cache-name }}-
          ${{ runner.os }}-build-
          ${{ runner.os }}-
```

Source: https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action